src/ja/worker/proxy/proxy.py

```diff
@@ -55,14 +61,27 @@ class WorkerServerProxy(IWorkerServerProxy):
     Implementation of the proxy for the central server used on the worker client.
     """
 
-    def register_self(self, work_machine_resources: WorkMachineResources) -> ServerResponse:
+    def _get_ssh_connection(self, ssh_config: SSHConfig) -> ISSHConnection:
         pass
 
-    def unregister_self(self) -> ServerResponse:
-        pass
+    def register_self(self, uid: str, work_machine_resources: WorkMachineResources) -> Response:
+        wm: WorkMachine = WorkMachine(uid, WorkMachineState.ONLINE, work_machine_resources)
+        register_command: RegisterWorkerCommand = RegisterWorkerCommand(wm)
+        response = self._ssh_connection.send_server_command(register_command)
+        return response
 
-    def notify_job_finished(self, uid: str) -> ServerResponse:
-        pass
+    # maybe rename to retire self or?
```

Comment:
I don't think the name matters much.
If you want to rename it please wait until Joe has merged his PR.